### ISSUE TITLE
Legacy java version

### ISSUE THREAD

**Issue by BigAdam2005** [bitcoinj/bitcoinj#1625]
Bitcoinj is using legacy java7. With java8 we can use lambdas and streams so library would be faster.

**Comment by schildbach** [comment]
FYI `using java.util.stream` won't be possible for a couple of years. Android has it only since Android 7.0 (N), API level 24. Raising the minSdk to 24 would cut out about half of users atm.

**Comment by msgilligan** [comment]
Is upgrading to Java 8 really (still) a goal for version 0.16?

**Comment by schildbach** [comment]
Good catch, I'll remove the milestone. It'll be another couple of years until we can use the streams API.

**Comment by msgilligan** [comment]
I would say that as of version 0.17 this issue is completed.  (Although we support Android API level 26 which I believe doesn't support _every_ Java 8 library feature, but it does support Streams, CompletableFuture, etc.)
